docs/user/managing-fleets.md

```diff
@@ -91,27 +91,29 @@ A fleet's device template contains a device specification that gets applied to a
 
 For example, you could specify in a fleet's device template that all devices in the fleet shall run the OS image `quay.io/flightctl/rhel:9.5`. The Flight Control service would then roll out this specification to all devices in the fleet and the Flight Control agents would update the devices accordingly. The same would apply to the other specification items described in [Managing Devices](managing-devices.md).
 
-However, it would be impractical if *all* of a fleet's devices had to have the *exact same specification*. Flight Control therefore allows templates to contain placeholders that get filled in based on a device's name or label values.
+However, it would be impractical if *all* of a fleet's devices had to have the *exact same specification*. Flight Control therefore allows templates to contain placeholders that get filled in based on a device's name or label values. The syntax for these placeholders matches that of [Go templates](https://pkg.go.dev/text/template), but you may only use simple text or actions (no conditionals or loops, for example). You may reference anything under a device's metadata such as `{{ .Metadata.Labels.key }}` or `{{ .Metadata.Name }}`.
 
-Here are some examples what you can do with placeholders in device templates:
+We also provide some helper functions:
 
-* You can label devices by their deployment stage (say, `stage: testing` and `stage: production`) and then use the label with the key `stage` as placeholder when referencing the OS image to use (say, `quay.io/myorg/myimage:latest-{{ .Labels["stage"] }}`) or when referencing a folder with configuration in a Git repository.
-* You can label devices by deployment site (say, `site: factory-berlin` and `site: factory-madrid`) and then use the label with the key `site` as parameter when referencing the secret with network access credentials in Kubernetes.
+* `toUpper`: Change to upper case. For example, `{{ toUpper .Metadata.Name }}`.
```

Comment:
Should we support and prefer the pipeline syntax?

If there are multiple actions, the pipeline syntax feels cleaner and more intuitive, e.g.:
```
{{ .Metadata.Labels.stage | default "testing" | lower | replace " " "-" }}
```